internal/liqonet/tunnel-operator/natmapping_operator_test.go

```diff
@@ -0,0 +1,269 @@
+package tunneloperator
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/util/slice"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/liqotech/liqo/apis/net/v1alpha1"
+)
+
+const (
+	oldIP1    = "10.0.1.2"
+	oldIP2    = "10.0.1.5"
+	newIP1    = "10.0.2.4"
+	newIP2    = "10.0.3.7"
+	DNAT      = "DNAT"
+	timeout   = time.Second * 10
+	interval  = time.Millisecond * 250
+	namespace = "test"
+)
+
+var request = reconcile.Request{}
+
+var _ = Describe("NatmappingOperator", func() {
+	BeforeEach(func() {
+		request.Namespace = namespace
+		// Names to resources will be given in Its
+		nm1 = &v1alpha1.NatMapping{
+			ObjectMeta: v1.ObjectMeta{
+				Namespace: namespace,
+			},
+			Spec: v1alpha1.NatMappingSpec{
+				ClusterID:       clusterID1,
+				ClusterMappings: make(v1alpha1.Mappings),
+			},
+		}
+		nm2 = &v1alpha1.NatMapping{
+			ObjectMeta: v1.ObjectMeta{
+				Namespace: namespace,
+			},
+			Spec: v1alpha1.NatMappingSpec{
+				ClusterID:       clusterID2,
+				ClusterMappings: make(v1alpha1.Mappings),
+			},
+		}
+	})
+	AfterEach(func() {
+		err := k8sClient.DeleteAllOf(context.Background(), &v1alpha1.NatMapping{}, &client.DeleteAllOfOptions{
+			ListOptions: client.ListOptions{
+				Namespace: namespace,
+			},
+		})
+		Expect(err).To(BeNil())
+	})
+	Context("If a natmapping resource is deleted", func() {
```

Comment:
The operator does not add finalizers to the resources, hence when a resource is deleted the operator just can not fetch it and will receive an `notFoundError`. This test case is the same as the one in the context `Reconcile on a resource that does not exist`